crates/re_types_builder/src/objects.rs

```diff
@@ -68,28 +68,34 @@ impl Objects {
             for field in &obj.fields {
                 let virtpath = &field.virtpath;
                 if let Some(field_type_fqname) = field.typ.fqname() {
                     let field_obj = &this[field_type_fqname];
-                    if obj.kind == ObjectKind::Archetype {
-                        assert!(field_obj.kind == ObjectKind::Component,
-                            "{virtpath}: Field {:?} (pointing to an instance of {:?}) is part of an archetypes but is not a component. Only components are allowed as fields on an Archetype.",
-                            field.fqname, field_type_fqname
-                        );
-                    } else {
-                        assert!(field_obj.kind == ObjectKind::Datatype,
-                            "{virtpath}: Field {:?} (pointing to an instance of {:?}) is part of a Component or Datatype but is itself not a Datatype. Only Archetype fields can be Components, all other fields have to be primitive or be a datatypes.",
-                            field.fqname, field_type_fqname
-                        );
+                    match obj.kind {
+                        ObjectKind::Datatype | ObjectKind::Component => {
+                            if field_obj.kind != ObjectKind::Datatype {
+                                reporter.error(virtpath, &obj.fqname, format!("Points to an instance of {field_type_fqname:?}) is part of a Component or Datatype but is itself not a Datatype. Only archetype fields can be components, all other fields have to be primitive or be a datatypes."));
+                            }
+                        }
+                        ObjectKind::Archetype => {
+                            if field_obj.kind != ObjectKind::Component {
+                                reporter.error(virtpath, &obj.fqname, format!("Points to an instance of {field_type_fqname:?}) is part of an archetypes but is not a component. Only components are allowed as fields on an archetype."));
+                            }
+                        }
+                        ObjectKind::SpaceView => {
+                            if field_obj.kind != ObjectKind::Archetype {
+                                reporter.error(virtpath, &obj.fqname, format!("Points to an instance of {field_type_fqname:?}) is part of an space view but is not an archetype. Only archetypes are allowed as fields of a space view's properties."));
+                            }
+                        }
                     }
                 } else {
                     // Note that we *do* allow primitive fields on components for the moment. Not doing so creates a lot of bloat.
-                    assert!(obj.kind != ObjectKind::Archetype,
-                        "{virtpath}: Field {:?} is a primitive field of type {:?}. Only Components are allowed on Archetypes.",
-                        field.fqname, field.typ);
+                    if obj.kind == ObjectKind::Archetype && obj.kind == ObjectKind::SpaceView {
```

Comment:
```suggestion
                    if obj.kind == ObjectKind::Archetype || obj.kind == ObjectKind::SpaceView {
```
this looks like a logical error.